### PR TITLE
OSD-14675: Initializing repository

### PR DESCRIPTION
**Pull request by MitaliBhalla**
This PR serves the purpose of initializing the repository , please note this is a part of initialization and hence more like a scaffolding. The following have been added.
1. main.go (with a sample function)
2. go.mod
3. go.sum
4. openapi
5. makefile
6. Readme
7. Unit test

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	go test -v ./...
+	@echo Run \'make test-cover\' to see the test coverage report
+
+test-cover:
+	go test -cover -coverprofile=coverage.out ./...
+	go tool cover -html=coverage.out
+
+generate:
+	go generate -v ./...
+
```
Comment (supreeth7) [inline Makefile:98]: Please add this target as well.
```suggestion
dev-certs:
	bash -c 'openssl req \
		-x509 \
		-out localhost.crt \
		-keyout localhost.key \
		-newkey rsa:4096 \
		-nodes \
		-sha256 \
		-subj "/CN=localhost" \
		-extensions san \
		-config <( \
			echo "[req]"; \
			echo "distinguished_name=req"; \
			echo "[san]"; \
			echo "subjectAltName=DNS:localhost")'
ifeq ($(UNAME_S),Darwin)
	security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain localhost.crt
endif
ifeq ($(UNAME_S),Linux)
	cp localhost.crt /etc/pki/tls/certs/
	cp localhost.crt /etc/pki/ca-trust/source/anchors/
	update-ca-trust extract
endif

```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Backplane-API
+
+## What it does 
+
+Backplane is a management service which consolidates all access to OSD and ROSA clusters and cloud provider resources.Backplane handles authentication and authorization through OCM profiles, practices least privilege access to those OSD/ROSA clusters and cloud resources, and provides auditing capabilities for compliance and security. It is basically responsible for managing all OSD related things by serving as the central entry point.
```
Comment (supreeth7) [inline README.md]:
```suggestion
Backplane is a management service which consolidates all access to OSD and ROSA clusters and cloud provider resources. Backplane handles authentication and authorization through OCM profiles, practices least privilege access to those OSD/ROSA clusters and cloud resources, and provides auditing capabilities for compliance and security.
```

File: openapi/openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      description: Optional extended description in CommonMark or HTML.
+      responses:
+        '200':    # status code
+          description: A JSON array of test resourses
+          content:
+            application/json:
+              schema: 
+                type: array
+                items: 
+                  type: string
```
Comment (supreeth7) [inline openapi/openapi.yaml]: You should move the entire OpenAPI schema.
Also, please add a new line at the end of file.

File: main.go [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+	"net/http"
+
+	"github.com/gorilla/mux"
+)
+
+func main() {
+	fmt.Println("This is entrypoint for backplane-api")
+	mainRouter := mux.NewRouter()
+	mainRouter.NotFoundHandler = http.NotFoundHandler()
+}
```
Comment (supreeth7) [inline main.go]: This function isn't really doing anything at the moment.
You can create a simple handler and add a unit test for that.

File: openapi/openapi.yaml (hunk 3 above)
Comment (supreeth7) [inline openapi/openapi.yaml]: Please add a new line at EOF.

File: pkg/sampleTest.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package pkg
+
+import (
+	"fmt"
+)
+
+func sampleFunc() {
+	fmt.Println("Hello world")
+}
+
```
Comment (supreeth7) [inline pkg/sampleTest.go]: Please add a new line at EOF.

**Review by supreeth7 — COMMENTED** [review]
Thank you for working on this.
I've added a few comments, please take a look.

**Comment by supreeth7** [comment]
Also, please add the `Dockerfile` as well.

File: main.go (hunk 4 above)
Comment (MitaliBhalla) [reply main.go]: Updated, thanks

**Comment by MitaliBhalla** [comment]
@supreeth7 Thanks for the review, I have updated and incorporated changes, can you please take a look again? Thanks.

**Comment by supreeth7** [comment]
I'll setup the CI tomorrow and merge this.

**Comment by supreeth7** [comment]
/hold

Can you add `build-in-container` `build-static: clean`, `build: clean`, `build-image` targets to the Makefile as well?

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+			echo "[san]"; \
+			echo "subjectAltName=DNS:localhost")'
+ifeq ($(UNAME_S),Darwin)
+	security add-trusted-cert -d -r trustRoot -k /Library/Keychains/System.keychain localhost.crt
+endif
+ifeq ($(UNAME_S),Linux)
+	cp localhost.crt /etc/pki/tls/certs/
+	cp localhost.crt /etc/pki/ca-trust/source/anchors/
+	update-ca-trust extract
+endif
```
Comment (supreeth7) [inline Makefile]: This is duplicated twice in this file.

Comment (MitaliBhalla) [reply Makefile]: Thanks, updated. Apologies for the duplication.

**Comment by supreeth7** [comment]
/hold

Now that the CI is working, please fix the tests.

**Comment by supreeth7** [comment]
For cache error during lint: Add `GOCACHE=/tmp/` to the Makefile perhaps.

File: Dockerfile [hunk 7]
```diff
@@ -4,8 +4,6 @@ FROM registry.access.redhat.com/ubi8/ubi AS builder
 
 RUN yum install -y ca-certificates git go-toolset make
 ENV PATH="/root/go/bin:${PATH}"
-ADD https://password.corp.redhat.com/RH-IT-Root-CA.crt /etc/pki/ca-trust/source/anchors/
-RUN update-ca-trust extract
```
Comment (supreeth7) [inline Dockerfile]: This shouldn't have been removed just to mitigate the failing tests, instead you could try doing:
```
RUN curl -sfL https://password.corp.redhat.com/RH-IT-Root-CA.crt \
    -o /etc/pki/ca-trust/source/anchors/RH-IT-Root-CA.crt ;\
    update-ca-trust
```

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+.PHONY: lint-in-container
+lint-in-container: build-image
+	$(RUN_IN_CONTAINER_CMD) "go mod download && make lint"
+
+getdeps:
+	@command -v golangci-lint 1>/dev/null || (echo "Installing golangci-lint" && curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(shell go env GOPATH)/bin $(GOLANGCI_LINT_VERSION))
+
+.PHONY: lint
+lint: getdeps
+	export GOLANGCI_LINT_CACHE=/tmp
```
Comment (supreeth7) [inline Makefile]: Can you set this as a variable under L#11?
That would be:
```
GOLANGCI_LINT_VERSION=v1.50.1
GOLANGCI_LINT_CACHE=/tmp
```

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,134 @@
+IMAGE_REGISTRY?=quay.io
+IMAGE_REPOSITORY?=app-sre
+IMAGE_NAME?=backplane-api
+VERSION=$(shell git rev-parse --short=7 HEAD)
+UNAME_S := $(shell uname -s)
+HOME=$(mktemp -d)
```
Comment (supreeth7) [inline Makefile]:
```suggestion
HOME=$(shell mktemp -d)
```

**Comment by MitaliBhalla** [comment]
/lgtm
Testing for label to see if problem similar to cli repo suffices.

**Pull request merged**